src/OpenTelemetry.Abstractions/Trace/AttributeValue.cs

```diff
@@ -80,4 +105,74 @@ public static IAttributeValue<double> DoubleAttributeValue(double doubleValue)
             Func<double, T> doubleFunction,
             Func<object, T> defaultFunction);
     }
+
+    abstract class AttributeValueBoolean : AttributeValue {
```

Comment:
for boolean you can have two classes - one for True and one for False. And cache them to avoid allocations